piff/input.py

```diff
@@ -624,6 +656,18 @@ def _makeStarsFromImage(image_kwargs, cat_kwargs, wcs, chipnum,
             if g is not None:
                 logger.debug("Adding Poisson noise to weight map according to gain=%f",g)
                 star = star.addPoisson(gain=g)
+
+
+            if max_edge_frac is not None and max_edge_frac < 1:
+                flux = np.sum(star.image.array)
+                try:
+                    flux_extra = np.sum(star.image.array[edge_mask])
+                except IndexError:
+                    logger.info("Star at position %f,%f overlaps the edge of the image and max_edge_frac cut is set, skipping this star.")                    
+                    continue
+                if flux_extra / flux > max_edge_frac:
+                    continue
```

Comment:
Need a logger.warning here also to say skipping this star because x% of flux is near the edge.

Reply:
Minor point, but you already calculated flux_frac = flux_extra/flux, so no need to recalculate it here.